File: src/main/java/org/chousik/commands/validators/ScriptFileValidator.java
package org.chousik.commands.validators;

import org.chousik.exception.ArgumentError;

import java.io.File;

public class ScriptFileValidator {
    public void valid(String fileName) throws ArgumentError {
        File scriptFile = new File(fileName);
        if (!scriptFile.exists()) {
            throw new ArgumentError("Неверное имя файла");
        }
        if (!scriptFile.canRead()) {
            throw new ArgumentError("Файл недоступен к чтению");
        }
    }
}

File: src/main/kotlin/commands/validators/ScriptFileValidator.kt
package org.chousik.commands.validators

import org.chousik.exception.ArgumentError
import java.io.File

class ScriptFileValidator {
    fun valid(fileName: String?) {
        val scriptFile = File(fileName!!)
        if (!scriptFile.exists()) {
            throw ArgumentError("Неверное имя файла")
        }
        if (!scriptFile.canRead()) {
            throw ArgumentError("Файл недоступен к чтению")
        }
    }
}